pyomo/contrib/doe/doe.py

```diff
@@ -1149,24 +1201,36 @@ def det_general(m):
             )
             return m.det == det_perm
 
-        if self.Cholesky_option:
+        if self.Cholesky_option and self.objective_option == ObjectiveLib.det:
             m.cholesky_cons = pyo.Constraint(
                 m.regression_parameters, m.regression_parameters, rule=cholesky_imp
             )
             m.Obj = pyo.Objective(
                 expr=2 * sum(pyo.log(m.L_ele[j, j]) for j in m.regression_parameters),
                 sense=pyo.maximize,
             )
-        # if not cholesky but determinant, calculating det and evaluate the OBJ with det
+
         elif self.objective_option == ObjectiveLib.det:
+            # if not cholesky but determinant, calculating det and evaluate the OBJ with det
+            m.det = pyo.Var(initialize=np.linalg.det(fim), bounds=(small_number, None))
             m.det_rule = pyo.Constraint(rule=det_general)
             m.Obj = pyo.Objective(expr=pyo.log(m.det), sense=pyo.maximize)
-        # if not determinant or cholesky, calculating the OBJ with trace
+
         elif self.objective_option == ObjectiveLib.trace:
+            # if not determinant or cholesky, calculating the OBJ with trace
+            m.trace = pyo.Var(initialize=np.trace(fim), bounds=(small_number, None))
             m.trace_rule = pyo.Constraint(rule=trace_calc)
             m.Obj = pyo.Objective(expr=pyo.log(m.trace), sense=pyo.maximize)
+            # m.Obj = pyo.Objective(expr=m.trace, sense=pyo.maximize)
+
         elif self.objective_option == ObjectiveLib.zero:
+            # add dummy objective function
             m.Obj = pyo.Objective(expr=0)
+        else:
+            # something went wrong!
+            raise ValueError(
```

Comment:
This should raise a `DeveloperError`.  Actually, is it even reachable?  I think the only way to get here is to add a new entry into the ObjectiveLib enum...